packages/openapi-ts/src/utils/write/transformers.ts

```diff
@@ -0,0 +1,328 @@
+      );
+    case 'reference': {
+      if (model.$refs.length !== 1) {
+        return [];
+      }
+      const refModels = getRefModels(props);
+
+      const { created, name: nameModelResponseTransformer } =
+        ensureModelResponseTransformerExists({ ...props, model: refModels[0] });
+
+      if (!created) {
+        return [];
+      }
+
+      return model.in === 'response'
+        ? [
+            compiler.convert.expressionToStatement({
+              expression: compiler.function.call({
+                functionName: nameModelResponseTransformer,
+                parameters: [dataVariableName],
+              }),
+            }),
+          ]
+        : compiler.transform.transformItem({
+            path: props.path,
+            transformerName: nameModelResponseTransformer,
+          });
+    }
+    // unsupported
+    default:
+      return [];
+  }
+};
+
+const generateResponseTransformer = ({
+  client,
+  name,
+  onNode,
+  onRemoveNode,
+  statements,
+}: Pick<TypesProps, 'client' | 'onNode' | 'onRemoveNode'> & {
+  name: string;
+  statements: Array<ts.Statement>;
+}) => {
+  const result = {
+    created: false,
+    name,
+  };
+
+  if (!statements.length) {
+    // clean up created type for response transformer if it turns out
+    // the transformer was never generated
+    unsetUniqueTypeName({
+      client,
+      name,
+    });
+    onRemoveNode?.();
+    return result;
+  }
+
+  const expression = compiler.types.function({
+    multiLine: true,
+    parameters: [
+      {
+        name: dataVariableName,
+      },
+    ],
+    statements: [
+      ...statements,
+      compiler.return.statement({
+        expression: ts.factory.createIdentifier(dataVariableName),
+      }),
+    ],
+  });
+  const statement = compiler.export.const({
+    expression,
+    name,
+    typeName: name,
+  });
+  onNode(statement);
+
+  return {
+    created: true,
+    name,
+  };
+};
+
+export const processResponseTransformers = async ({
+  client,
+  onNode,
+  onRemoveNode,
+}: Pick<TypesProps, 'client' | 'onNode' | 'onRemoveNode'>) => {
+  const config = getConfig();
+
+  for (const service of client.services) {
+    for (const operation of service.operations) {
+      const hasRes = operation.results.length;
+
+      if (!hasRes) {
+        continue;
+      }
+
+      const responses = getSuccessResponses(operation.results);
+
+      const nonVoidResponses = responses.filter(
+        (response) => !isVoidResponse(response),
+      );
+
+      if (!nonVoidResponses.length) {
+        continue;
+      }
+
+      if (nonVoidResponses.length > 1) {
+        if (config.debug) {
+          console.warn(
+            `⚠️ Transformers warning: route ${operation.method} ${operation.path} has ${responses.length} success responses. This is currently not handled and we will not generate a response transformer. Please open an issue if you'd like this feature https://github.com/hey-api/openapi-ts/issues`,
+          );
```

Comment:
I do think putting this behind a debug flag is inviting people to have weird issues and blame the package. If it fires it should be because a type IS wrong and that's something devs need to know up front